beam/beam_job_service.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+
+set -euxo pipefail
+
+readonly LOCAL_JAR_NAME='beam-runners-flink_2.11-job-server.jar'
+readonly SERVICE_INSTALL_DIR='/usr/lib/beam-job-service'
+readonly SERVICE_WORKING_DIR='/var/lib/beam-job-service'
+readonly SERVICE_WORKING_USER='yarn'
+
+readonly ARTIFACTS_DIR_METADATA_KEY='beam-artifacts-dir'
+readonly RELEASE_SNAPSHOT_URL_METADATA_KEY="beam-job-service-snapshot"
+readonly RELEASE_SNAPSHOT_URL_DEFAULT="http://repo1.maven.org/maven2/org/apache/beam/beam-runners-flink_2.11-job-server/2.6.0/beam-runners-flink_2.11-job-server-2.6.0.jar"
+
+readonly BEAM_IMAGE_ENABLE_PULL_METADATA_KEY="beam-image-enable-pull"
+readonly BEAM_IMAGE_ENABLE_PULL_DEFAULT=false
+readonly BEAM_IMAGE_VERSION_METADATA_KEY="beam-image-version"
+readonly BEAM_IMAGE_VERSION_DEFAULT="master"
+readonly BEAM_IMAGE_REPOSITORY_KEY="beam-image-repository"
+readonly BEAM_IMAGE_REPOSITORY_DEFAULT="apache.bintray.io/beam"
+
+
+readonly START_FLINK_YARN_SESSION_METADATA_KEY='flink-start-yarn-session'
+# Set this to true to start a flink yarn session at initialization time.
+readonly START_FLINK_YARN_SESSION_DEFAULT=true
+
+function is_master() {
+  local role="$(/usr/share/google/get_metadata_value attributes/dataproc-role)"
+  if [[ "$role" == 'Master' ]] ; then
+    true
+  else
+    false
+  fi
+}
+
+function get_artifacts_dir() {
+  /usr/share/google/get_metadata_value "attributes/${ARTIFACTS_DIR_METADATA_KEY}" \
+    || echo "gs://$(/usr/share/google/get_metadata_value "attributes/dataproc-bucket")/beam-artifacts"
```

Comment:
If I'm not mistaken, `attributes/dataproc-bucket` points to the Dataproc _staging_ bucket for this cluster rather than the Dataproc init action bucket (which we should hard code if we want it).

Reply:
Yes, this is correct.  My intention was to use the Dataproc staging bucket as the default artifacts bucket if none is supplied by the user.  The alternative is to make this metadata key required at creation time, so that we have somewhere user-created to store the artifacts.  I don't think we would want to store customer artifacts in the dataproc init action bucket.

It was sort of an independent design choice, but the staging bucket seemed like a good place to put them if no artifact directory was specified.  It also helps with the cleanup of integration tests.

Reply:
Sorry, not "Dataproc" init action bucket, but the bucket associated with this init action's artifacts. Seems reasonable to keep it like this for now, though I'm not sure how/when the staging buckets are cleaned up if created by Dataproc.